sites/kit.svelte.dev/src/lib/search/SearchBox.svelte

```diff
@@ -0,0 +1,416 @@
+<script>
+	import { onMount } from 'svelte';
+	import { Icon } from '@sveltejs/site-kit';
+	import flexsearch from 'flexsearch';
+	import { afterNavigate } from '$app/navigation';
+	import { searching, query, recent } from './stores.js';
+	import { focusable_children, trap } from '../actions/focus.js';
+
+	let modal;
+
+	let results = [];
+	let backspace_pressed;
+
+	let index;
+	let lookup;
+
+	onMount(async () => {
+		const response = await fetch('/content.json');
+		const { blocks } = await response.json();
+
+		index = new flexsearch.Index({
+			tokenize: 'forward'
+		});
+
+		lookup = new Map();
+
+		let time = Date.now();
+		for (const block of blocks) {
+			const title = block.breadcrumbs[block.breadcrumbs.length - 1];
+			lookup.set(block.href, {
+				title,
+				href: block.href,
+				breadcrumbs: block.breadcrumbs.slice(0, -1),
+				content: block.content
+			});
+			index.add(block.href, `${title} ${block.content}`);
+
+			// poor man's way of preventing blocking
+			if (Date.now() - time > 25) {
+				await new Promise((f) => setTimeout(f, 0));
+				time = Date.now();
+			}
+		}
+
+		update();
+	});
+
+	afterNavigate(() => {
+		// TODO this also needs to apply when only the hash changes
+		// (should before/afterNavigate fire at that time? unclear)
+		$searching = false;
+	});
+
+	function update() {
+		results = (index ? index.search($query) : []).map((href) => lookup.get(href));
+	}
+
+	function escape(text) {
+		return text.replace(/</g, '&lt;').replace(/>/g, '&gt;');
+	}
+
+	function excerpt(content, query) {
+		const index = content.toLowerCase().indexOf(query.toLowerCase());
+		if (index === -1) {
+			return content.slice(0, 100);
+		}
+
+		const prefix = index > 20 ? `…${content.slice(index - 15, index)}` : content.slice(0, index);
+		const suffix = content.slice(
+			index + query.length,
+			index + query.length + (80 - (prefix.length + query.length))
+		);
+
+		return (
+			escape(prefix) +
+			`<mark>${escape(content.slice(index, index + query.length))}</mark>` +
+			escape(suffix)
+		);
+	}
+
+	/** @param {string} href */
+	function navigate(href) {
+		$recent = [href, ...$recent.filter((x) => x !== href)];
+		$searching = false;
+	}
+
+	$: if ($searching) update();
+
+	$: recent_searches = lookup ? $recent.map((href) => lookup.get(href)).filter(Boolean) : [];
+</script>
+
+<svelte:window
+	on:keydown={(e) => {
+		if (e.code === 'KeyK' && (navigator.platform === 'MacIntel' ? e.metaKey : e.ctrlKey)) {
+			e.preventDefault();
+			$query = '';
+			$searching = !$searching;
+		}
+
+		if (e.code === 'Escape') {
+			$searching = false;
```

Comment:
When closing the modal, we need to move focus somewhere so it's not lost (though only if the user didn't select an option, i.e. they're not navigating). The candidates would either be 1) the search input in the nav bar (it's where focus initially was, but could be confusing to non-sighted users, since doing so would read out "Search input" again) or 2) the skip link in the nav (basically resetting to the top of the page)

Reply:
I tried focusing the skip link but it meant that it became visible...

<img width="655" alt="image" src="https://user-images.githubusercontent.com/1162160/152870270-6406cd0c-1ea3-416e-af66-4c9e8e07b36a.png">

...which I think we probably want to avoid. I just pushed a change that focuses the `<body>` instead, maybe that's ok?